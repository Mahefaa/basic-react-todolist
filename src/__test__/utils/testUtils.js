export const TASK_TODO_STATUS = "TODO";
export const TASK_DONE_STATUS = "DONE";