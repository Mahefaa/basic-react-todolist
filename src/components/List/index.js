import style from './index.module.css'
import ListItem from "../ListItem";


export default function List({list, setList, status}) {
    const handleCheck = (e) => {
        if (e.target.checked) {
            let updatedList = list.map((task, id) => {
                if (id === parseInt(e.target.value)) {
                    return {
                        ...task,
                        status: "DONE"
                    };
                }
                return task;
            });
            setList(updatedList);
        }
    }
    return (
        <div className={style.list}>
            <h1 className={style.title}>{status}</h1>
            <div className={style.list__items} data-testid = {`list-${status}`}>
                {
                    list.map((task, index) => (
                        task.status === status &&
                        <ListItem task={task} index={index} handleCheck={handleCheck}/>
                    ))
                }
            </div>
        </div>
    )
}