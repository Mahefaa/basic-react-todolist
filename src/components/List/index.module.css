.list {
    display: flex;
    flex-direction: column;
    min-height: 200px;
    width: 20vw;
    word-break: break-all;
}
.list__items{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 10px;
    overflow-y: scroll;
}
.title {
    text-align: center;
    font-size: 1.5em;
    border-bottom: 2px solid black;
    padding: 5px;
}