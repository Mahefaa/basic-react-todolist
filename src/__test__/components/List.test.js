import {cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";
import List from "../../components/List";
import {TASK_DONE_STATUS, TASK_TODO_STATUS} from "../utils/testUtils";

afterEach(cleanup);
test("List component is rendered correctly", () => {
    const list = [
        {id: 1, name: "wash dishes", status: TASK_TODO_STATUS},
        {id: 2, name: "eat dinner", status: TASK_TODO_STATUS},
        {id: 3, name: "prepare dinner", status: TASK_DONE_STATUS},
        {id: 4, name: "eat breakfast", status: TASK_DONE_STATUS}
    ];
    const tree = renderer
        .create(<List list={list} setList={null} status={TASK_TODO_STATUS}/>)
        .toJSON();

    expect(tree).toMatchSnapshot();
})