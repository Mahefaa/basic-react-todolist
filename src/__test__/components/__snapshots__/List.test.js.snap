// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component is rendered correctly 1`] = `
<div
  className="list"
>
  <h1
    className="title"
  >
    TODO
  </h1>
  <div
    className="list__items"
    data-testid="list-TODO"
  >
    <div>
      <input
        data-testid="TODO-0"
        onClick={[Function]}
        type="checkbox"
        value={0}
      />
      <span
        className="task__name todo"
      >
        wash dishes
      </span>
    </div>
    <div>
      <input
        data-testid="TODO-1"
        onClick={[Function]}
        type="checkbox"
        value={1}
      />
      <span
        className="task__name todo"
      >
        eat dinner
      </span>
    </div>
  </div>
</div>
`;
